src/main/java/frc/robot/OI.java

```diff
@@ -0,0 +1,232 @@
+      return new JoystickButton(kJoystick, kAlignBackwardButton.getButtonID());
+    }
+
+    public static JoystickButton getIntakeButton() {
+      kIntakeButton.setButtonAction("Intake");
+      return new JoystickButton(kJoystick, kIntakeButton.getButtonID());
+    }
+
+    public static JoystickButton getOuttakeButton() {
+      kOuttakeButton.setButtonAction("Outtake");
+      return new JoystickButton(kJoystick, kOuttakeButton.getButtonID());
+    }
+  }
+
+  public static final class Operator {
+    private static enum Button {
+      A(1),
+      B(2),
+      X(3),
+      Y(4),
+      LB(5), // Left Bumper
+      RB(6), // Right Bumper
+      Back(7),
+      Start(8),
+      LJ(9), // Left Joystick Button
+      RJ(10), // Right Joystick Button
+      POVUP(0),
+      POVDOWN(180),
+      POVLEFT(270),
+      POVRIGHT(90);
+
+      private final int buttonID;
+      private String buttonAction;
+
+      Button(int ID) {
+        this.buttonID = ID;
+        this.buttonAction = "";
+      }
+
+      private int getButtonID() {
+        return this.buttonID;
+      }
+      ;
+
+      private String getButtonAction() {
+        return this.buttonAction;
+      }
+
+      private void setButtonAction(String name) {
+        this.buttonAction = name;
+      }
+    };
+
+    private static final Joystick kJoystick = new Joystick(OI.kOperatorJoystickPort);
+    private static final Button kExampleButton =
+        Button.LB; // Toggles intake mode between cone and cube
+
+    public static JoystickButton getExampleButton() {
+      kExampleButton.setButtonAction("Example");
+      return new JoystickButton(kJoystick, kExampleButton.getButtonID());
+    }
+  }
+
+  public static void putControllerButtons() {
+    ShuffleboardLayout driverButtonsLayout =
+        Shuffleboard.getTab("Controls")
+            .getLayout("Driver Buttons", BuiltInLayouts.kList)
+            .withSize(2, 5)
+            .withPosition(0, 0)
+            .withProperties(Map.of("Label position", "HIDDEN")); // hide labels for Variables;
+
+    ShuffleboardLayout operatorButtonsLayout =
+        Shuffleboard.getTab("Controls")
+            .getLayout("Operator Buttons", BuiltInLayouts.kList)
+            .withSize(2, 5)
+            .withPosition(2, 0)
+            .withProperties(Map.of("Label position", "HIDDEN")); // hide labels for Variables;
+
+    for (Driver.Button button : Driver.Button.values()) {
+      driverButtonsLayout.add(
+          String.valueOf(button.getButtonID()),
+          "Button " + button.toString() + ": " + button.getButtonAction());
+    }
+
+    for (Operator.Button button : Operator.Button.values()) {
+      operatorButtonsLayout.add(
+          String.valueOf(button.getButtonID() + Operator.Button.values().length),
+          "Button " + button.toString() + ": " + button.getButtonAction());
+    }
+  }
+
+  public static class ControlCurve {
+    private double ySaturation; // Maximum output, in percentage of possible output
```

Comment:
It looks like this value is set when the class is instantiated, then never updated again. If that is correct, you should declare it `final`. The same goes for your function parameters below and elsewhere. `final` not only causes a variable to be unchangeable, it also _requires_ that it be set, issuing a compilation error when it's not, and also treats the variables differently during the compilation process. If you're interested, [here](https://stackoverflow.com/questions/27369584/what-does-the-jvm-do-with-a-final-variable-in-a-class) is a good SO Q&A about this difference.